tests/meta/backup/test_meta_backup.py

```diff
@@ -1,272 +1,193 @@
-    backup_api.maximum_flowcells_ondisk.return_value = False
 
-    # AND no flowcell has been specified
-    mock_flowcell = None
+    # AND no flow cell has been specified
+    mock_flow_cell = None
 
-    result = backup_api.fetch_flowcell(mock_flowcell, dry_run=False)
+    result = backup_api.fetch_flowcell(mock_flow_cell, dry_run=False)
 
-    # THEN no flowcell will be fetched and a log message indicates that no flowcells have been
+    # THEN no flow cell will be fetched and a log message indicates that no flow cells have been
     # requested
     assert result is None
-    assert "no flowcells requested" in caplog.text
+    assert "no flow cells requested" in caplog.text
 
 
-@mock.patch("cg.meta.backup.BackupApi.pop_flowcell")
-@mock.patch("cg.meta.backup.BackupApi.maximum_flowcells_ondisk")
-@mock.patch("cg.meta.backup.BackupApi.check_processing")
-@mock.patch("cg.apps.pdc")
+@mock.patch("cg.meta.backup.backup.BackupApi.pop_flowcell")
+@mock.patch("cg.meta.backup.backup.BackupApi.check_processing")
+@mock.patch("cg.meta.backup.pdc")
 @mock.patch("cg.store")
-def test_fetch_flowcell_retrieve_next_flowcell(
+def test_fetch_flow_cell_retrieve_next_flow_cell(
     mock_store,
     mock_pdc,
     mock_check_processing,
-    mock_maximum_flowcells_ondisk,
     mock_pop_flowcell,
+    cg_context,
     caplog,
 ):
-    """tests the fetch_flowcell method of the backup API"""
+    """tests the fetch_flow cell method of the backup API"""
 
     caplog.set_level(logging.INFO)
 
-    # GIVEN we check if a flowcell needs to be retrieved from PDC
-    backup_api = BackupApi(
-        mock_store, mock_pdc, max_flowcells_on_disk=1250, root_dir="/path/to/root_dir"
-    )
+    # GIVEN we check if a flow cell needs to be retrieved from PDC
+    backup_api = BackupApi(mock_store, mock_pdc, root_dir=cg_context.backup.root.dict())
 
-    # WHEN no flowcells is specified, but a flowcell in status-db has the status "requested"
+    # WHEN no flow cell is specified, but a flow cell in status-db has the status "requested"
     mock_flowcell = None
     mock_pop_flowcell.return_value = mock_store.add_flowcell(
         status="requested",
     )
     backup_api.check_processing.return_value = True
-    backup_api.maximum_flowcells_ondisk.return_value = False
 
     result = backup_api.fetch_flowcell(mock_flowcell, dry_run=False)
 
-    # THEN the process to retrieve the flowcell from PDC is started
+    # THEN the process to retrieve the flow cell from PDC is started
     assert "retrieving from PDC" in caplog.text
 
-    # AND when done the status of that flowcell is set to "retrieved"
+    # AND when done the status of that flow cell is set to "retrieved"
     assert (
-        f'Status for flowcell {mock_pop_flowcell.return_value.name} set to "retrieved"'
+        f'Status for flow cell {mock_pop_flowcell.return_value.name} set to "retrieved"'
         in caplog.text
     )
     assert mock_pop_flowcell.return_value.status == "retrieved"
```

Comment:
In line 272 I'd like to explicitly use `"requested"` instead of an Enum constant. The FlowCellStatus Enum is part of the code, and the values in that Enum are required values in the Flowcell table in cg:
```
-- auto-generated definition
create table flowcell
(
    id             int auto_increment
        primary key,
    name           varchar(32)                                                        not null,
    sequencer_type enum ('hiseqga', 'hiseqx', 'novaseq')                              null,
    sequencer_name varchar(32)                                                        null,
    sequenced_at   datetime                                                           null,
    archived_at    datetime                                                           null,
    status         enum ('ondisk', 'processing', 'removed', 'requested', 'retrieved') null,
    updated_at     datetime                                                           null,
    constraint name
        unique (name)
)
    charset = latin1;
```

If we assert against the Enum value, and we change that enum value, the unit test will pass and the code will fail when commiting an invalid value to the table.

What I mean is that `assert mock_flow_cell.status == FlowCellStatus.RETRIEVED` will happily pass if the Enum says 
```
class FlowCellStatus(StrEnum):
    ONDISK: str = "ondisk"
    REMOVED: str = "removed"
    REQUESTED: str = "requested"
    PROCESSING: str = "processing"
    RETRIEVED: str = "i_should_actually_fail_the_unit_test"
```